hs_core/task_utils.py

```diff
@@ -101,34 +81,16 @@ def get_or_create_task_notification(task_id, status='progress', name='', payload
                 obj.status = status
             obj.save()
 
-        if obj.status == "progress":
-            if not task_exists(task_id):
-                obj.status = 'failed'
-                obj.save()
-
         return {
             'id': task_id,
+            'task_id': task_id,
```

Comment:
why need to add another key task_id with the same value as id?

Reply:
I mentioned it slack to you, but it does warrant a comment in the code.  It's for backwards compatibility with rest calls.